network/src/commonTest/kotlin/com/wire/kalium/api/tools/json/api/asset/AssetApiTest.kt

```diff
@@ -0,0 +1,162 @@
+                assertPost()
+                assertJson()
+                assertNoQueryParams()
+                assertAuthorizationHeaderExist()
+                assertPathEqual(PATH_PUBLIC_ASSETS)
+            }
+        )
+
+        // When
+        val assetApi: AssetApi = AssetApiImp(httpClient)
+        val response = assetApi.uploadAsset(assetMetadata, encryptedData)
+
+        // Then
+        assertTrue(response.isSuccessful())
+        assertEquals(response.value, VALID_ASSET_UPLOAD_RESPONSE.serializableData)
+    }
+
+    @Test
+    fun givenAnInvalidAssetUploadApiRequest_whenCallingTheAssetUploadApiEndpoint_theRequestShouldContainAnError() = runTest {
+        // Given
+        val assetMetadata = AssetMetadata("image/jpeg", true, AssetRetentionType.ETERNAL, "md5-hash")
+        val encryptedData = ByteArray(16)
+        val httpClient = mockAuthenticatedHttpClient(
+            INVALID_ASSET_UPLOAD_RESPONSE.rawJson,
+            statusCode = HttpStatusCode.BadRequest,
+            assertion = {
+                assertPost()
+                assertJson()
+                assertNoQueryParams()
+                assertAuthorizationHeaderExist()
+                assertPathEqual(PATH_PUBLIC_ASSETS)
+            }
+        )
+
+        // When
+        val assetApi: AssetApi = AssetApiImp(httpClient)
+        val response = assetApi.uploadAsset(assetMetadata, encryptedData)
+
+        // Then
+        assertTrue(response is NetworkResponse.Error)
+        assertTrue(response.kException is KaliumException.InvalidRequestError)
+    }
+
+    @Test
+    fun givenAValidAssetDownloadApiRequest_whenCallingTheAssetDownloadApiEndpoint_theRequestShouldBeConfiguredCorrectly() = runTest {
+        // Given
+        val downloadedAsset = ByteArray(16)
+        val assetKey = "3-1-e7788668-1b22-488a-b63c-acede42f771f"
+        val assetToken = "assetToken"
+        val apiPath = "$PATH_PUBLIC_ASSETS/$assetKey"
+        val httpClient = mockAuthenticatedHttpClient(
+            responseBody = downloadedAsset,
+            statusCode = HttpStatusCode.OK,
+            assertion = {
+                assertGet()
+                assertNoQueryParams()
+                assertAuthorizationHeaderExist()
+                assertHeaderExist(HEADER_ASSET_TOKEN)
+                assertPathEqual(apiPath)
+            }
+        )
+
+        // When
+        val assetApi: AssetApi = AssetApiImp(httpClient)
+        val response = assetApi.downloadAsset(assetKey, assetToken)
+
+        // Then
+        assertTrue(response.isSuccessful())
+
+        // We can't assert that the response object is exactly the same object as the provided on the Mocked HttpClient because MockEngine
+        // from Ktor seems to copy internally the content of the original ByteArray into a different ByteArray, therefore they are not
+        // technically the same object. That's why we are just doing a simple array size check
+        assertEquals(response.value.size, downloadedAsset.size)
```

Comment:
```suggestion
        assertEquals(String(response.value, charset = Charsets.UTF_8), String(downloadedAsset, charset = Charsets.UTF_8))
```
this should solve the problem